astropy/table/table.py

```diff
@@ -782,8 +791,8 @@ def _new_from_slice(self, slice_):
         """Create a new table as a referenced slice from self."""
 
         table = self.__class__(masked=self.masked)
-        table.meta.clear()
-        table.meta.update(deepcopy(self.meta))
+        if self.meta:
+            table.meta = self.meta.copy()  # Shallow copy for slice
```

Comment:
For a slice it seems that doing at least a shallow (key) copy is a good idea instead of a direct object reference.  The most common case is a single-level table meta (e.g. from a FITS file), and the dict copy is pretty fast.